### PR TITLE
[v6r20] Add tools for Echo

### PR DESCRIPTION
**Pull request by chaen**
This is a preliminary work needed to be able to use Echo (new RAL storage).
Replaces https://github.com/DIRACGrid/DIRAC/pull/3870

@andresailer  Can you have a look in particular to the FTS part ? Normally it should not change anything but just to make sure..

BEGINRELEASENOTES

*DMS
NEW: FTS submissions can use any third party protocol
NEW: Storage plugin for Echo (gsiftp+root)
FIX: replace deprecated calls to the gfal2 API
NEW: Generic implementation for retrieving space occupancy on storage

ENDRELEASENOTES

### CODE REVIEW

**Comment by chaen** [comment]
FYI, I tested it in our certification machine, and this change does not break anything for LHCb. So please approve and merge ASAP

File: Resources/Storage/EchoStorage.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
+""" :mod: EchoStorage
```
Comment (andresailer) [inline Resources/Storage/EchoStorage.py]: sphinx autodoc will take care of creating the documentation strings for modules, classes etc. So to not cause warnings, please just put
```
"""Echo module based on the GFAL2_StorageBase class."""
```

File: Resources/Storage/EchoStorage.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+
+class EchoStorage(GFAL2_StorageBase):
+  """ .. class:: EchoStorage
+
+    Interface to the Echo storage.
+
+    This plugin will work with both gsiftp and root protocol.
+    According to the RAL admins, Echo best use is done:
+      * using gsiftp for all WAN transfers, and LAN write.
```
Comment (andresailer) [inline Resources/Storage/EchoStorage.py]: Another sphinx syntax thing:
The bullet list should be separated by an empty line, but indented to the same level as the preceding paragraph.

**Comment by andresailer** [comment]
I cherry-picked the commit for the FTS3Agent, things seem to still run.

File: Resources/Storage/EchoStorage.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    """
+    # # init base class
+    super(EchoStorage, self).__init__(storageName, parameters)
+    self.srmSpecificParse = False
+
+    self.log = gLogger.getSubLogger("EchoStorage")
+
+    self.pluginName = 'Echo'
+
+    # Because Echo considers '<host>/lhcb:prod' differently from '<host>//lhcb:prod' as it normaly should be
```
Comment (fstagni) [inline Resources/Storage/EchoStorage.py]: *normally

File: Resources/Storage/StorageBase.py [hunk 4]
```diff
@@ -429,9 +435,40 @@ def _isInputURL(self, url):
...
+    # Build the URL for the occupancyLFN:
+    occupancyLFN = kwargs['occupancyLFN']
+    res = self.constructURLFromLFN(occupancyLFN)
+    if not res['OK']:
+      return res
+    occupancyURL = res['Value']
+
+    try:
+
+      # download the file localy
```
Comment (fstagni) [inline Resources/Storage/StorageBase.py]: *locally

**Comment by chaen** [comment]
Thanks, review implemented

**Pull request merged**